Modules/_io/bytesio.c

```diff
@@ -440,6 +440,12 @@ bytesio_truncate(bytesio *self, PyObject *args)
 
     CHECK_CLOSED(self);
 
+    if (PyErr_WarnPy3k_WithFix("BytesIO.truncate() does not shift the file pointer",
```

Comment:
At this point do we know that `truncate(0)` has been called? Won't this warning also trigger with e.g. `truncate(1)`?